Reword "name" and "about" issue template frontmatter (#7570) (#7938)
github/docs: .github/ISSUE_TEMPLATE/improve-the-site.md

---
name: Improve the docs.github.com site

> about: Make a suggestion to improve the technical implementation of docs.github.com.

Author: Could also be:
```suggestion
about: Make a suggestion or report a problem about the technical implementation of docs.github.com.
```
